graphs/frequent_pattern_graph_miner.py

```diff
@@ -0,0 +1,195 @@
+'''
+FP-GraphMiner - A Fast Frequent Pattern Mining Algorithm for Network Graphs
+
+A novel Frequent Pattern Graph Mining algorithm,
+FP-GraphMiner, that compactly represents a set of network graphs as a Frequent Pattern Graph (or FP-Graph).
+This graph can be used to efficiently mine frequent subgraphs including maximal frequent subgraphs and maximum common subgraphs.
+
+URL:https://www.researchgate.net/publication/235255851_FP-GraphMiner_-_A_Fast_Frequent_Pattern_Mining_Algorithm_for_Network_Graphs
+
+'''
+from typing import List
+
+edge_array=[
+    ['ab-e1','ac-e3','ad-e5','bc-e4','bd-e2','be-e6','bh-e12','cd-e2','ce-e4','de-e1','df-e8','dg-e5','dh-e10','ef-e3','eg-e2','fg-e6','gh-e6','hi-e3'],
+    ['ab-e1','ac-e3','ad-e5','bc-e4','bd-e2','be-e6','cd-e2','de-e1','df-e8','ef-e3','eg-e2','fg-e6'],
+    ['ab-e1','ac-e3','bc-e4','bd-e2','de-e1','df-e8','dg-e5','ef-e3','eg-e2','eh-e12','fg-e6','fh-e10','gh-e6'],
+    ['ab-e1','ac-e3','bc-e4','bd-e2','bh-e12','cd-e2','df-e8','dh-e10'],
+    ['ab-e1','ac-e3','ad-e5','bc-e4','bd-e2','cd-e2','ce-e4','de-e1','df-e8','dg-e5','ef-e3','eg-e2','fg-e6']
+    ]
+
+def get_distinct_edge(edge_array: List[List[str]]) -> List[str]:
+    '''
+    Return Distinct edges from edge array of multiple graphs
+    '''
+    distinct_edge=set()
+
+    for i,row in enumerate(edge_array):    
+        for j,item in enumerate(row):
+            distinct_edge.add(item[0])
+
+    return list(distinct_edge)
+
+def get_bitcode(edge_array,distinct_edge):
+    '''
+    Return bitcode of distinct_edge
+    '''
+    bitcode=['0' for i in enumerate(edge_array)]
+    
+    for i,row in enumerate(edge_array):
+        for j,item in enumerate(row):
+            if distinct_edge in item[0]:
+                bitcode[i]='1'
+                break
+
+            
+    return bitcode
+
+def get_frequency_table(edge_array):
+    '''
+    Returns Frequency Table,cluster,nodes,support
+    '''
+    distinct_edge=get_distinct_edge(edge_array) 
+    frequency_table=dict()
+    
+    for i,item in enumerate(distinct_edge):
+        bit=get_bitcode(edge_array,item)
+        bt=''.join(bit)
+        s=bt.count('1')
+        frequency_table[item]=[s,bt]
+    '''
+    Store [Distinct edge, WT(Bitcode), Bitcode] in Descending order
+    '''

```

Comment:
However, further down in the function, normal comments should be used instead...
```suggestion
    # Store [Distinct edge, WT(Bitcode), Bitcode] in Descending order
```